### DISCUSSION TITLE
Vaillant aerotherm missing vallues

### DISCUSSION THREAD

**Discussion by eyk107** [john30/ebusd#1196] [General]
Hi everyone,

I' m using a v5 ebusd adapter in HA and I've got everything running but a few information are missing (see attachment).
My serial number is: 0010022066
Could someone please help me, I'm a little bit lost on this issue.
Thank you in advance
[ebusd.log](https://github.com/john30/ebusd/files/14251813/ebusd.log)

**Comment by JonesGW-MGD** [comment]
Hi,

Having started on the ebusd journey about four weeks ago, my experience is it depends how much time you want to invest!  Ending up with anything other than 'out of the box' data involves effort.

First point is - you may never get all the commands decoded.  The project relies on configuration files that map cryptic and proprietary ebus messages (telegrams) and decode them into a useable format. From what I understand, it has mostly been done by trial and error and experimentation. So if there isn't a configuration file for your device, or the entry in the configuration file for a particular message command byte doesn't match your setup you'll get unreadable messages until that situation changes.

Second point is - what are you missing?  Rather than worry about unknown commands, maybe start with assessing what you do get and what you'd like to have that you don't.

But - and there is no short response to this - If you want more than you have, and you think it should be available, the first step is to make sure you're in the best possible starting position.  

(I'll give you the layman's version, because that is what I am!) 

When ebusd starts, it interrogates the ebus network to work out what devices on are on the bus.  It then loads configuration files it believes are the best match for those devices.  So a starting point is to ensure it loads as many config files as it can.  

Your log extract look like it only loads 15.ctlv2.csv.  '15' indicates it's the slave address of a controller - in your case a SENSOComfort.  (It might not be immediately obvious but most devices usually have two addressess - one as being a 'master' and one as a 'slave'. (There are some devices that are only slaves).  Master/Slave addresses are hex values offset by 5 - so in your log, you can see messages starting with '10' - that's your SENSOComfort' in master mode; if you see a message with xx15 where xx is another address, that is another master device sending a message to the SENSOComfort)

(Maybe other files load later in the logs, I'd expect you to have 08.hmu.csv loaded as well if you have a heat pump).

There are fairly easy ways to establish what files loaded - which we can talk about if you're interested in going further!

Some devices may not have config files - It looks like maybe you have a Hydraulic Connection Unit? (maybe a VWZ MEH?) as device 71/76 and there may not be files for this device.

It's not all bad though - because maybe the HCU doesn't send anything useful you can't find elsewhere.

For example:  Generally, most useful data will come from your 10/15 device. (In your case the SENSOComfort)

In any event, once you have the most possible config files loaded, in my view the next easiest thing to do is to potentially correct/change some of the command bytes in the config files by experimentation.  This requires first though that you clone the files locally and then you can amend them.  

For example, I found on my gas boiler that 

```
r;wi,,d.00_PartloadHcKW,d.00 heating partload,,,,"6C00",,,power,,,Adjustable partial heat load (Auto: Product automatically adjusts max. partial load to current system demand)
```

originally had the command byte set in the default file to 0704 and that gave unknown for my set up.

How did I find 6C00?  I looked at other config files of the same type that had variations for this parameter.  

So there is experimentation that you can try to 'improve' the default config files.

You can find by searching for your devices on this forum and the discussion forum for the configuration files themselves: 

https://github.com/john30/ebusd-configuration/issues 

that people may have done some of the hard work already and shared changes to the default repository files that work but are not part of the official repository.

If that still doesn't give you what you want you can try to decode the messages - the wiki has a 'how to'. - you may or may not be successful in this if you try - but no doubt others will be interested in your result!!

So..... maybe start with seeing what you don't have that you'd like rather than fixing something that might not be a problem would be my best advice.  (Happy to help further if I can).

Good luck!
